apis/python/tests/test_dataframe.py

```diff
@@ -536,6 +536,27 @@ def make_multiply_indexed_dataframe(tmp_path, index_column_names: List[str]):
             "A": [12, 13, 14, 15],
             "throws": None,
         },
+        {
```

Comment:
one thing this does not test is a case where:
* an indexed column allows negative int values
* and the user slices negatively

Eg., a DataFrame with an indexed column of type int64, actual column values of [-10..10], and a read that slices out `[-2:2]`, etc.

Unless I am misreading the code in `make_multiply_indexed_dataframe`, all numeric coordinates are currently positive, which means there may still be dragons in coordinate values <= 0

Reply:
Added.